src/main/java/org/mule/tools/apikit/model/MuleConfigBuilder.java

```diff
@@ -10,57 +10,66 @@
 
-  public MuleConfig build() {
-    return new MuleConfig(httpListenerConfigs, apiKitConfigs, flows);
+  public static MuleConfig fromDoc(Document muleConfigContent) {
+    return fromDoc(muleConfigContent, DEFAULT_HTTP_LISTENER_CONFIG_PERSISTED);
   }
 
-  public static MuleConfig fromDoc(Document muleConfigContent) {
-    HttpListenerConfigParser httpConfigParser = new HttpListenerConfigParser();
+  public static MuleConfig fromDoc(Document muleConfigContent, boolean httpListenerConfigPersisted) {
+    HttpListenerConfigParser httpConfigParser = new HttpListenerConfigParser(httpListenerConfigPersisted);
     APIKitConfigParser apiKitConfigParser = new APIKitConfigParser();
+    APIAutodiscoveryConfigParser apiAutodiscoveryConfigParser = new APIAutodiscoveryConfigParser();
+    ConfigurationPropertiesConfigParser configurationPropertiesConfigParser = new ConfigurationPropertiesConfigParser();
 
     List<HttpListenerConfig> httpListenerConfigs = httpConfigParser.parse(muleConfigContent);
     List<APIKitConfig> apikitConfigs = apiKitConfigParser.parse(muleConfigContent);
+    APIAutodiscoveryConfig apiAutodiscoveryConfig = apiAutodiscoveryConfigParser.parse(muleConfigContent);
+    ConfigurationPropertiesConfig configurationPropertiesConfig = configurationPropertiesConfigParser.parse(muleConfigContent);
+
     List<Flow> flowsInConfig = new ArrayList<>();
 
     for (Content content : muleConfigContent.getRootElement().getContent()) {
       if (content instanceof Element) {
         Element contentElement = (Element) content;
         if ("flow".equals(contentElement.getName())) {
           Optional<ApikitRouter> apikitRouter = getRouter(contentElement);
+          Flow flow = new Flow(contentElement);
           if (apikitRouter.isPresent()) {
             MainFlow mainFlow = new MainFlow(contentElement);
             mainFlow.setApikitRouter(apikitRouter.get());
-            flowsInConfig.add(mainFlow);
-          } else {
-            flowsInConfig.add(new Flow(contentElement));
+            flow = mainFlow;
           }
+          flowsInConfig.add(flow);
         }
       }
     }
-
-    return new MuleConfig(httpListenerConfigs, apikitConfigs, flowsInConfig, muleConfigContent);
+    return new MuleConfig(httpListenerConfigs, apikitConfigs, flowsInConfig, apiAutodiscoveryConfig,
+                          configurationPropertiesConfig, muleConfigContent);
   }
 
   public static MuleConfig fromStream(InputStream input) throws Exception {
+    return fromStream(input, DEFAULT_HTTP_LISTENER_CONFIG_PERSISTED);
+  }
+
+  public static MuleConfig fromStream(InputStream input, boolean httpListenerConfigPersisted) throws Exception {
     SAXBuilder builder = MuleConfigBuilder.getSaxBuilder();
     Document inputAsDocument = builder.build(input);
     input.close();
-    return fromDoc(inputAsDocument);
+    return fromDoc(inputAsDocument, httpListenerConfigPersisted);
```

Comment:
httpListenerConfigPersisted review this

Reply:
Yeah, this is very ugly. It seems that it comes from `HttpListenerConfig` and it has been there for a while, but we should avoid spreading this bad design all over the code.